lib/postcode_validation/use_case/format_validator.rb

```diff
@@ -0,0 +1,19 @@
+require_relative 'format_validators/nl_postcode_validator'
+require_relative 'format_validators/be_postcode_validator'
+require_relative 'format_validators/gb_postcode_validator'
+require_relative 'format_validators/fr_postcode_validator'
+
+class FormatValidator
+   def for country
+     case country
+     when 'NL'
+       NLPostcodeValidator.new
```

Comment:
We haven't fully explored the problem space. 

One example: What if the way to validate is not "regular".. I.e. requires a [Type-0, Type-1 or Type2 grammar to validate](https://en.wikipedia.org/wiki/Chomsky_hierarchy#The_hierarchy).

:man_shrugging: I actually don't know if this will ever happen, I generally reach for polymorphism by class hierarchies before I reach for metaprogramming.

Another reason to fragment is to think about the Single Responsibility Principle

There's an actor "GB Postcodes Policy" actor, an "FR Postcodes Policy" actor, etc.... If you make one class knowledgeable (or to put it another way, open to change by all those actors), you have violated the SRP.

By closing the use case class to change from those actors, you've made that class *stable*. Future programmers will be less likely to change that code, meaning less likely to introduce bugs there.